File: Day10-BinaryNumbers/Java/main.java
import java.util.*;

public class Solution {
    public static int mostConsecutiveOnes(int n) {
        // convert number->binary string->char array
        char[] binary = Integer.toBinaryString(n).toCharArray();
        
        // count of current sequence of consecutive ones
        int tmpCount = 0; 
        
        // running maximum count of consecutive ones for any section to left of tmpCount
        int maxCount = 0; 
        for(int i = 0; i < binary.length; i++){
            
            // reset to 0 if we hit a '0' char
            if(binary[i] == '0') {
                
                // set new max if needed
                if(tmpCount > maxCount){
                    maxCount = tmpCount;
                }
                    
                tmpCount = 0;
            }
            else { // current location is a section of consecutive 1's
                // increment tmpCount
                tmpCount =  tmpCount + 1; 
            }
        }
        // conditional is necessary here in case the string does not end with a 0
        return (tmpCount > maxCount) ? tmpCount : maxCount;
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        scan.close();

        System.out.println(mostConsecutiveOnes(n));
    }
}

File: Day10-BinaryNumbers/C++/main.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int maxCount = 0;
    int count = 0;
    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');
    vector<int> binary;
    vector<int>::iterator itr;
    while (n)
    {
        binary.insert(binary.begin(), n % 2);
        n /= 2;
    }
    for (itr = binary.begin(); itr != binary.end(); itr++)
    {
        if (*itr == 1)
        {
            count += 1;
        }
        else
            count = 0;
        if (count > maxCount)
        {
            maxCount = count;
        }
    }
    cout << maxCount;
    return 0;
}
